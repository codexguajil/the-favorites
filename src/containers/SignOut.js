import React, { Component } from 'react';
import { connect } from 'react-redux';
import { logOutUser } from '../actions';

export class SignOut extends Component {

   SignOut = () => {
    console.log('youre signed out')
    this.props.logOutUser();
  }

  render() {
    return (
      <button onClick={this.SignOut}>Log Out</button>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  logOutUser: (user) => dispatch(logOutUser())
})

export default connect(null, mapDispatchToProps)(SignOut);