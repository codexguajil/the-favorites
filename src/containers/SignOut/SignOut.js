import React, { Component } from 'react';
import { connect } from 'react-redux';
import { logOutUser } from '../../actions';

export class SignOut extends Component {

   SignOut = () => {
    this.props.logOutUser();
  }

  render() {
    return (
      <button onClick={this.SignOut}>Log Out</button>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  logOutUser: (user) => dispatch(logOutUser())
})

export default connect(null, mapDispatchToProps)(SignOut);