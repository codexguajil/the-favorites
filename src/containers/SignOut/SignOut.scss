.sign-out-btn {
  background: none;
  border: none;
  border-radius: 5px;
  color: hsla(360, 0%, 100%, 1);
  font-family: 'nav';
  font-size: 16px;
  height: 40px;
  width: 90px;
  &:hover {
    cursor: pointer;
  }
}