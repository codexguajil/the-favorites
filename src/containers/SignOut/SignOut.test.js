import React from 'react';
import SignOut from './SignOut'
import { mapDispatchToProps } from './SignOut'
import { shallow } from 'enzyme'

describe('SignOut', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <SignOut />
    )

    expect(wrapper).toMatchSnapshot()
  })

  describe('mapDispatchToProps', () => {
    it('should pass correct type when addFavoriteToStore is called', () => {
      const dispatch = jest.fn()

      mapDispatchToProps(dispatch).logOutUser()
      expect(dispatch.mock.calls[0][0]).toEqual({ type: 'LOG_OUT_USER' })
    });
  })
})