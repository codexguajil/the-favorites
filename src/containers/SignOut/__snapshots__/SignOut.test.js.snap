// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
